staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -56,9 +56,12 @@ var defaultSleepDuration = 10 * time.Second
 
 // TODO: validate impact of copying and investigate hashing
 type eventKey struct {
+	eventType           string
 	action              string
 	reason              string
+	note                string
```

Comment:
Adding note is less clear to me - why do we need it?

Reply:
I added it for now just to be consistent with the core API aggregation since `note` is the new field for `message`.

That said I think it makes sense if users emit events that have variable information in the note since they will most likely want to see the updated values right away. The problem is that this introduces back the potential for spammy events, but does it really makes sense to disregard the note value completely?

Reply:
ohh yeah, sorry, I thought we're keeping that in the reason, bu `reason` is the machine-readable thing

Yeah - I think the note should be here, and it's on us (reviewers/approvers of the code) to ensure, that things like time or sth like that is not written to the note.
Ideally, some static analysis would be useful, but we probably won't get to it any time in the near future.

So yeah - it seems fine.

Reply:
I think this was intentionally excluded from aggregation keys before, because of the scale problems we were getting with events

Reply:
That's true, but given that migration to new events is explicit (i.e. we need to touch every single place where event is generated), we can afford together with the migration, ensure that the `note` field is not artificially different for the same cases (e.g. it doesn't contain timestamp, current state or sth like that).

So the question is whether such discipline when reviewing this code is achievable.

Reply:
> So the question is whether such discipline when reviewing this code is achievable.

I'll defer to the folks who added the aggregation in the first place in response to production fires around event volume, but I suspect it is not

Reply:
I have a feeling that aggregating events with different notes will seem like a bug to the users. Let's imagine a scenario where users intentionally want to produce events with different notes. If they are only able to see the first note and not the following ones as they were aggregated to the first event, then it will seem like a bug to them.

The scale problems are understandable and concerning, but I think that the default event recorder should avoid introducing data loss. A potential option could be to allow tweaking the aggregation mechanism at the event recorder level and disregard the `note` field for all Kubernetes components.

Reply:
I'm a little confused about whether we're trying to preserve status quo on top of the new API version, or make changes.

Reply:
To be honest this PR was mostly a placeholder for me to remember that the aggregation needs to be fixed :sweat_smile:

I could start by only adding `eventType` and `reportingInstance` to preserve the status quo of the new API and then in a separate effort consolidate aggregation based on `note` to regain the same behavior as the core API, but while adding a new functionality to prevent scalability issues as I detailed above.

Reply:
Also, there shouldn't be anything preventing us from adding the protection mechanism to the core API event recorder so the aggregation logic would be the same in both APIs in the end.